pylearn2/linear/tests/test_cudnn.py

```diff
@@ -0,0 +1,146 @@
+"""
+Tests for the Cudnn code
+"""
+
+import theano
+from theano import tensor
+from theano.sandbox.cuda.dnn import dnn_available
+
+from pylearn2.linear.conv2d import Conv2D
+from pylearn2.linear.cudnn2d import Cudnn2D, make_random_conv2D
+from pylearn2.space import Conv2DSpace
+from pylearn2.utils import sharedX
+from pylearn2.testing.skip import skip_if_no_gpu
+
+import unittest
+from nose.plugins.skip import SkipTest
+
+import numpy as np
+
+
+class TestCudnn(unittest.TestCase):
+    """
+    Tests for the Cudnn code
+    """
+    def setUp(self):
+        """
+        Set up a test image and filter to re-use
+        """
+        skip_if_no_gpu()
+        if not dnn_available():
+            SkipTest('Skipping tests cause cudnn is not available')
+        self.image = np.random.rand(1, 1, 3, 3).astype(theano.config.floatX)
```

Comment:
So I tried to run the tests, this fails when floatX is float64 (which is the default).
A solution would be to force floatX to be float32, as is done in https://github.com/fvisin/pylearn2/blob/mlp_cudnn/pylearn2/linear/tests/test_conv2d_c01b.py#L25, and restore that in `tearDown`. I don't think you have to worry about the call to `use` or `unuse`, though.